File: backend/onGo-domain/src/main/kotlin/com/ongo/domain/activitylog/ActivityLog.kt
package com.ongo.domain.activitylog

import java.time.LocalDateTime

data class ActivityLog(
    val id: Long? = null,
    val userId: Long,
    val action: String,
    val entityType: String? = null,
    val entityId: Long? = null,
    val details: String = "{}",
    val ipAddress: String? = null,
    val userAgent: String? = null,
    val createdAt: LocalDateTime? = null,
)

File: frontend/src/api/activityLog.ts
import apiClient, { unwrapResponse } from './client'
import type { ResData } from '@/types/api'

export interface ActivityLogResponse {
  id: number
  userId: number
  action: string
  entityType: string | null
  entityId: number | null
  details: string
  ipAddress: string | null
  createdAt: string | null
}

export interface ActivityLogListResponse {
  logs: ActivityLogResponse[]
  totalElements: number
  page: number
  size: number
}

export const activityLogApi = {
  list(params?: { page?: number; size?: number; action?: string; entityType?: string; startDate?: string; endDate?: string }) {
    return apiClient.get<ResData<ActivityLogListResponse>>('/activity-logs', { params }).then(unwrapResponse)
  },
}
